clients/ts/signalr/src/Transports.ts

```diff
@@ -320,17 +326,37 @@ export class LongPollingTransport implements ITransport {
     public onclose: TransportClosed;
 }
 
-function getDataDetail(data: any): string {
+function getDataDetail(data: any, includeContent: boolean): string {
     let length: string = null;
     if (data instanceof ArrayBuffer) {
-        length = `Binary data of length ${data.byteLength}`;
+        length = `Binary data of length ${data.byteLength}.`;
+        if (includeContent) {
+            length += ` Content: ${formatArrayBuffer(data)}`;
+        }
     } else if (typeof data === "string") {
-        length = `String data of length ${data.length}`;
+        length = `String data of length ${data.length}.`;
+        if (includeContent) {
+            length += ` Content: ${data}`;
+        }
     }
     return length;
 }
 
-async function send(logger: ILogger, transportName: string, httpClient: HttpClient, url: string, accessTokenFactory: () => string, content: string | ArrayBuffer): Promise<void> {
+function formatArrayBuffer(data: ArrayBuffer): string {
+    const view = new Uint8Array(data);
+
+    // Uint8Array.map only supports returning another Uint8Array?
+    let str = "";
+    view.forEach((num) => {
+        const pad = num < 16 ? "0" : "";
+        str += `0x${pad}${num.toString(16)} `;
```

Comment:
Should we worry about perf for large binary data, e.g. someone sending a 10meg message. This loop will create a lot of strings.

Reply:
Yeah, I was thinking that. For right now, that's somewhat mitigated by the fact that it's off by default.